src/main/scala/strawman/collection/Factories.scala

```diff
@@ -47,6 +43,18 @@ object IterableFactory {
 
 }
 
+trait IterableFactoryWithBuilder[+CC[_]] extends IterableFactory[CC] {
```

Comment:
What would you think of `StrictIterableFactory` instead of `IterableFactoryWithBuilder`?